libraries/common/cs/devtools-utils.js

```diff
@@ -60,16 +60,33 @@ export const getOrderedTopBlockColumns = (separateOrphans, workspace) => {
   let maxWidths = {};
 
   if (separateOrphans) {
-    let topComments = w.getTopComments();
+    if (w.getTheme) {
+      // new Blockly
+      const blocks = w.getAllBlocks();
+      for (const block of blocks) {
+        for (const icon of block.getIcons()) {
+          if (icon.commentBubble) {
+            const comment = icon.commentBubble;
+            const right = comment.getRelativeToSurfaceXY().x + comment.getSize().width;
+
+            const root = comment.getSourceBlock().getRootBlock();
+            const left = root.getRelativeToSurfaceXY().x;
+            maxWidths[root.id] = Math.max(right - left, maxWidths[root.id] || 0);
+          }
+        }
+      }
+    } else {
+      let topComments = w.getTopComments();
 
-    for (const comment of topComments) {
-      if (comment.setVisible) {
-        autoPositionComment(comment);
-        let right = comment.getBoundingRectangle().bottomRight.x;
+      for (const comment of topComments) {
+        if (comment.block_) {
```

Comment:
This is for old Blockly, but I changed it to make it more obvious what the if statement is for (it checks if the comment is a block comment).